packages/react/src/components/Geo/AmplifyMap/index.tsx

```diff
@@ -0,0 +1,75 @@
+import type { ICredentials } from '@aws-amplify/core';
+import { Amplify, Auth } from 'aws-amplify';
+import maplibregl from 'maplibre-gl';
+import { AmplifyMapLibreRequest } from 'maplibre-gl-js-amplify';
+import React, { useEffect, useState } from 'react';
+import ReactMapGL from 'react-map-gl';
+import type { MapProps, TransformRequestFunction } from 'react-map-gl';
+
+/**
+ * The `AmplifyMap` component uses [react-map-gl](https://visgl.github.io/react-map-gl/) and
+ * [maplibre-gl-js](https://visgl.github.io/react-map-gl/) to provide an interactive map using
+ * [Amplify Geo APIs](https://docs.amplify.aws/lib/geo/getting-started/q/platform/js/) powered by
+ * [Amazon Location Service](https://aws.amazon.com/location/). Since `AmplifyMap` is a wrapper of the
+ * [react-map-gl default Map](https://visgl.github.io/react-map-gl/docs/api-reference/map), it accepts the same
+ * properties.
+ *
+ * @example
+ * // Basic usage of AmplifyMap:
+ * function App() {
+ *   return <AmplifyMap />
+ * }
+ */
+export const AmplifyMap = ({ style, ...props }: MapProps) => {
```

Comment:
This kind of ties in with my comment below on spread props - it may be valuable to export our own `AmplifyMapProps` interface that removes any props that we explicitly use in our `AmplifyMap` component, e.g.
```
export interface AmplifyMapProps extends Omit<MapProps, 'transformRequest' | 'mapLib' | 'mapStyle'>;
```

Reply:
This is a good callout and what I had originally written. I'm in agreement and can implement it this way unless there are other thoughts here.

Reply:
See my comment above; we can omit the `transformRequest` property while allowing `mapLib` and `mapStyle` to be overwritten.